src/main/java/com/google/collegeplanner/servlets/PlannerServlet.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.collegeplanner.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns the mutli-semester plan for the given courses.*/
+@WebServlet("/planner")
+public class PlannerServlet extends HttpServlet {
+  /**
+   * Organizes courses into the given number of semesters
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I think this is better suited with `doPost`. GET requests  typically always serve the same the same content every time it's called (like website pages) but POST requests are more complicated and do things that you don't want people to be able to do by just typing in the url in their browser (like logging in, submitting forms, etc).  This servlet will eventually do more complicated things so I think POST is better here.

Reply:
+1, great callout.